update created date if it isn't set by user

mathewmeconry opened the pull request:
if the object is created before a sync operation. It could be that the created is to old to be accepted by the server

mathewmeconry committed:
update created date if it isn't set by user

if the object is created before a sync operation. It could be that the created is to old to be accepted by the server

bouzuya commented:
Thank you for your pull request.
But I don't think that `getWSSEHeader()` should not update `this._created`.
Because `UserToken` is an immutable object.

Alternative plan:

```
UsernameToken.prototype.newToken = function() {
  // new nonce & new created
  return new UsernameToken({
    username: this._username,
    password: this._password,
    sha1encoding: this._sha1encoding
  });
};
```

mathewmeconry committed:
add newToken function

made the changes requested in the pull request

mathewmeconry committed:
remove unused function

removed the function isCreated which returned the _isCreated boolean which isn't in use anymore

mathewmeconry commented:
sorry for this late answer.

The alternative plan is the better way. I implemented it like this
